compiler/rustc_typeck/src/check/fn_ctxt/checks.rs

```diff
@@ -585,7 +583,9 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 )) = errors.iter().next()
                 {
                     let expected_ty = expected_input_tys[*arg_idx];
-                    let provided_ty = final_arg_types[*arg_idx].map(|ty| ty.0).unwrap();
+                    let provided_ty = final_arg_types[*input_idx]
```

Comment:
Should be using `input_idx`. Not sure if this is "correct" (though I think it is), but regardless, if there are more expected args than provided args, then this might be out of bounds.